### PR TITLE
fix: add helper functions to project for pkging

### PR DESCRIPTION
**Pull request by peternhale**
### What does this PR do?
Add some additional functions in SfProject for packaging needs
### What issues does this PR fix or reference?
@W-11748608@

### CODE REVIEW

File: src/sfProject.ts [hunk 1]
```diff
@@ -544,6 +604,17 @@ export class SfProject {
     return packageDir ? packageDir.name : undefined;
   }
 
+  /**
+   * Returns the package directory that has the given property whose value equals the given value.
+   *
+   * @param property
+   * @param value
+   */
+  public getPackageDirectoryWithProperty(property: string, value: string): Optional<NamedPackageDir> {
```
Comment (shetzel) [inline src/sfProject.ts:613]: Is this a common usecase?  This is essentially a filter.  I.e., `getPackageDirectories().filter(pd => pd.prop === value)` where the filter returns a single match.

File: src/sfProject.ts [hunk 2]
```diff
@@ -656,6 +727,34 @@ export class SfProject {
 
     return this.projectConfig;
   }
+
+  public async hasPackageAliases(): Promise<boolean> {
+    return await this.getSfProjectJson().hasPackageAliases();
```
Comment (shetzel) [inline src/sfProject.ts:732]:
```suggestion
    return this.getSfProjectJson().hasPackageAliases();
```
based on: https://jakearchibald.com/2017/await-vs-return-vs-return-await/

File: src/sfProject.ts [hunk 3]
```diff
@@ -656,6 +727,34 @@ export class SfProject {
...
+  }
+  /**
+   * Returns a read-only list of `packageDirectories` within sfdx-project.json, first reading
+   * and validating the file if necessary. i.e. modifying this array will not affect the
+   * sfdx-project.json file.
+   */
+  public getPackageAliases(): Nullable<Dictionary<string>> {
```
Comment (shetzel) [inline src/sfProject.ts:739]: this is already defined above

**Review by shetzel — CHANGES_REQUESTED** [review]

File: src/sfProject.ts [hunk 4]
```diff
@@ -302,21 +303,79 @@ export class SfProjectJson extends ConfigFile {
...
   public hasMultiplePackages(): boolean {
-    return this.getContents().packageDirectories && this.getContents().packageDirectories.length > 1;
+    return this.getContents()?.packageDirectories?.length > 1;
+  }
+
+  /**
+   * Has at least one package alias defined in the project.
+   */
+  public async hasPackageAliases() {
+    return Object.keys(this.getContents().packageAliases || {}).length > 0;
```
[the remark this answers is not in the document]
Comment (peternhale) [reply src/sfProject.ts:327]: Just keeping things aligned with the current complexity of the class. I do imagine more to follow.

File: src/sfProject.ts (hunk 1 above)
Comment (peternhale) [reply src/sfProject.ts:613]: Right now the function I wanted to replace from pkg utils is used twice, both are looking up which package dir is the default. I will remove this for now and we can  remove refs to `getConfigPackageDirectory` in pkg utils with calls to project.getDefaultPackage

File: src/sfProject.ts (hunk 3 above)
Comment (peternhale) [reply src/sfProject.ts:739]: The other function is on SfProjectJson. Once again, mirroring the current impl

**Review by shetzel — APPROVED** [review]

**Review by mdonnalley — APPROVED** [review]

**Pull request merged**